EE_endpoints.md

```diff
@@ -0,0 +1,130 @@
+### Introduction
+Each request comes in through an endpoint. Endpoints handle the request up to the
+point of passing it off the a [Router](http://www.phoenixframework.org/docs/routing).
+
+Each request begins and ends it's lifecycle inside your application in an endpoint.
+
+The sample [Hello
+Phoenix](https://github.com/phoenix-examples/hello_phoenix) Phoenix application starts up our HelloPhoenix.Endpoint as a supervised process.
+This is where Supervision from Erlang and Elixir enter this framework.
+
+Below is from
+[`lib/hello_phoenix.ex`](https://github.com/phoenix-examples/hello_phoenix/blob/master/lib/hello_phoenix.ex#L9-L16):
+```elixir
+ # ...
+ children = [¬
+   # Start the endpoint when the application starts¬
+   supervisor(HelloPhoenix.Endpoint, []),¬
+   # ... Other supervised processes like Ecto Repo or
+   # worker processes
+   supervisor(HelloPhoenix.Repo, []),
+   supervisor(HelloPhoenix.QueueWorker, [])
+  ]
+  # ...
+```
+
+These lines declare that our application is composed of multiple process trees.
+
+### Endpoint Contents
+An endpoint often includes plugs that are useful throughout all routes of
+your application. At the end of an endpoint file we can define multiple routers that would
+handle the actual routes.
+
+Let's cover some of the elements of an Endpoint layer by layer from outside
+moving inward by looking at the endpoint.ex in the [Hello
+Phoenix](https://github.com/phoenix-examples/hello_phoenix) sample Phoenix
+application.
+
+An `Endpoint` is an Elixir module like the `Controller` and `Router`.
```

Comment:
I don't think we need this line, we can just go straight into the code from the section above.